lib/spack/docs/module_file_support.rst

```diff
@@ -0,0 +1,704 @@
+===============================
+Integration with module systems
+===============================
+
+The use of module systems to manage user environment in a controlled way
+is a common practice at HPC centers that is often embraced also by individual
+programmers on their development machines. To blend in this common practice
+Spack provides integration with `Environment Modules
+<http://modules.sourceforge.net/>`_ ,  `LMod
+<http://lmod.readthedocs.io/en/latest/>`_ and `Dotkit <https://computing.llnl.gov/?set=jobs&page=dotkit>`_ by:
+
+ * generating module files after a successful installation
+ * providing commands that can leverage the spec syntax to manipulate modules
+
+In the following you will see how to activate shell support for commands in Spack
+that requires it, and discover what benefits this may bring with respect to deal
+directly with automatically generated module files.
+
+.. note::
+
+   If your machine does not already have a module system installed,
+   we advice to use either Environment Modules or LMod. See :ref:`InstallEnvironmentModules`
+   for more details.
+
+----------------------
+Activate shell support
+----------------------
+
+You can enable shell support by sourcing the appropriate setup file
+in the ``$SPACK_ROOT/share/spack`` directory.
+For ``bash`` or ``ksh`` users:
+
+.. code-block:: bash
+
+   . ${SPACK_ROOT}/share/spack/setup-env.sh
+
+For ``csh`` and ``tcsh`` instead:
+
+.. code-block:: csh
+
+   source $SPACK_ROOT/share/spack/setup-env.csh
```

Comment:
If these are intended to be command line commands, you should use "console" for the code block type and start them with a "$" to match the rest of the documentation. "bash" and "csh" are only intended for bash and csh scripts.